File: cpp/test/graph/test_dijkstra.cc
#include <vector>

#include <gtest/gtest.h>

#include "../../graph/dijkstra.cc"

using namespace std;

TEST(DijkstraTest, TestSingleNode) {
  vector<Node> graph = {{0}};
  ASSERT_EQ(Dijkstra(1, graph, 0, 0), 0);
}

TEST(DijkstraTest, TestPaths) {
  vector<Node> graph = {{10}, {20}, {30}, {40}};
  graph[0].AddAdj(1, 2);
  graph[0].AddAdj(1, 2);
  graph[1].AddAdj(3, 3);
  graph[2].AddAdj(3, 1);

  ASSERT_EQ(Dijkstra(4, graph, 0, 3), 5);
  ASSERT_EQ(Dijkstra(4, graph, 2, 3), 1);
  ASSERT_EQ(Dijkstra(4, graph, 1, 2), -1);
}

File: python/test/graph/test_dijkstra.py
import unittest

from graph.dijkstra import Node, Dijkstra


class DijkstraTest(unittest.TestCase):
    def test_single_node(self):
        n = Node(1)
        self.assertEqual(Dijkstra(n, 1), 0)
        self.assertEqual(Dijkstra(n, 2), -1)

    def test_paths(self):
        nodes = [Node(i) for i in xrange(10, 50, 10)]
        nodes[0].add_adj(nodes[1], 2)
        nodes[0].add_adj(nodes[2], 5)
        nodes[1].add_adj(nodes[3], 3)
        nodes[2].add_adj(nodes[3], 1)

        self.assertEqual(Dijkstra(nodes[0], 40), 5)
        self.assertEqual(Dijkstra(nodes[0], 50), -1)
        self.assertEqual(Dijkstra(nodes[2], 40), 1)
